### ISSUE TITLE
Issue while running bundle install. Permission denied websocket-driver-0.7.5

### ISSUE THREAD

**Issue by dareosK** [rubygems/rubygems#5721]
# Error Report

## Questions

Please fill out answers to these questions, it'll help us figure out
why things are going wrong.

- **What did you do?**

  I ran the command `/Users/dareosk/.gem/ruby/2.6.0/bin/bundle install`

- **What did you expect to happen?**

  I expected Bundler to install the dependencies

- **What happened instead?**

I got the following error

- **Have you read our issues document, https://github.com/rubygems/rubygems/blob/master/bundler/doc/contributing/ISSUES.md?**
this gives me a 404

- **What else did you try?**
clearing the cache with: `rm -rf ~/.bundle/cache`
upgrading rubygems and bundler as proposed here

  ...

## Backtrace

```
Errno::EACCES: Permission denied @ dir_s_mkdir - /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/extensions/x86_64-darwin-19/2.6.0/websocket-driver-0.7.5
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/2.6.0/fileutils.rb:239:in `mkdir'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/2.6.0/fileutils.rb:239:in `fu_mkdir'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/2.6.0/fileutils.rb:217:in `block (2 levels) in mkdir_p'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/2.6.0/fileutils.rb:215:in `reverse_each'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/2.6.0/fileutils.rb:215:in `block in mkdir_p'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/2.6.0/fileutils.rb:200:in `each'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/2.6.0/fileutils.rb:200:in `mkdir_p'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/rubygems/ext/builder.rb:217:in `write_gem_make_out'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/rubygems/ext/builder.rb:145:in `build_error'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/rubygems/ext/builder.rb:179:in `rescue in build_extension'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/rubygems/ext/builder.rb:168:in `build_extension'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/rubygems/ext/builder.rb:205:in `block in build_extensions'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/rubygems/ext/builder.rb:202:in `each'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/rubygems/ext/builder.rb:202:in `build_extensions'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/rubygems/installer.rb:851:in `build_extensions'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.22/lib/bundler/rubygems_gem_installer.rb:64:in `build_extensions'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.22/lib/bundler/rubygems_gem_installer.rb:26:in `install'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.22/lib/bundler/source/rubygems.rb:182:in `install'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.22/lib/bundler/installer/gem_installer.rb:67:in `install'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.22/lib/bundler/installer/gem_installer.rb:18:in `install_from_spec'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.22/lib/bundler/installer/parallel_installer.rb:186:in `do_install'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.22/lib/bundler/installer/parallel_installer.rb:177:in `block in worker_pool'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.22/lib/bundler/worker.rb:62:in `apply_func'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.22/lib/bundler/worker.rb:57:in `block in process_queue'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.22/lib/bundler/worker.rb:54:in `loop'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.22/lib/bundler/worker.rb:54:in `process_queue'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.22/lib/bundler/worker.rb:88:in `block (2 levels) in create_threads'
```

## Environment

```
Bundler       2.2.22
  Platforms   ruby, x86_64-darwin-19
Ruby          2.6.6p146 (2020-03-31 revision 67876) [x86_64-darwin19]
  Full Path   /Users/dareosk/.rbenv/versions/2.6.6/bin/ruby
  Config Dir  /Users/dareosk/.rbenv/versions/2.6.6/etc
RubyGems      3.3.17
  Gem Home    /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0
  Gem Path    /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0:/Users/dareosk/.gem/ruby/2.6.0
  User Home   /Users/dareosk
  User Path   /Users/dareosk/.gem/ruby/2.6.0
  Bin Dir     /Users/dareosk/.rbenv/versions/2.6.6/bin
OpenSSL       
  Compiled    OpenSSL 1.1.1n  15 Mar 2022
  Loaded      OpenSSL 1.1.1n  15 Mar 2022
  Cert File   /Users/dareosk/.rbenv/versions/2.6.6/openssl/ssl/cert.pem
  Cert Dir    /Users/dareosk/.rbenv/versions/2.6.6/openssl/ssl/certs
Tools         
  Git         2.27.0
  RVM         not installed
  rbenv       rbenv 1.2.0
  chruby      not installed
```

## Bundler Build Metadata

```
Built At          2021-07-06
Git SHA           0bdd3e8e71
Released Version  true
```

### Gemfile

```ruby
source 'https://rubygems.org'
git_source(:github) { |repo| "https://github.com/#{repo}.git" }

ruby '2.6.6'

# Bundle edge Rails instead: gem 'rails', github: 'rails/rails', branch: 'main'
gem 'rails', '~> 6.1.3', '>= 6.1.3.2'
# Use postgresql as the database for Active Record
gem 'pg', '~> 1.1'
# Use Puma as the app server
gem 'puma', '~> 5.0'
# Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
gem 'jbuilder', '~> 2.7'
# Use Redis adapter to run Action Cable in production
# gem 'redis', '~> 4.0'
# Use Active Model has_secure_password
# gem 'bcrypt', '~> 3.1.7'

# Use Active Storage variant
# gem 'image_processing', '~> 1.2'

# Reduces boot times through caching; required in config/boot.rb
gem 'bootsnap', '>= 1.4.4', require: false

# Use Rack CORS for handling Cross-Origin Resource Sharing (CORS), making cross-origin AJAX possible
gem 'rack-cors'
gem 'devise'
gem 'devise-jwt'
gem 'pundit'

group :development, :test do
  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
  gem 'dotenv-rails'
  gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
end

group :development do
  gem 'listen', '~> 3.3'
  # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
  gem 'spring'
end

# Windows does not include zoneinfo files, so bundle the tzinfo-data gem
gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
```

### Gemfile.lock

```
GEM
  remote: https://rubygems.org/
  specs:
    actioncable (6.1.4)
      actionpack (= 6.1.4)
      activesupport (= 6.1.4)
      nio4r (~> 2.0)
      websocket-driver (>= 0.6.1)
    actionmailbox (6.1.4)
      actionpack (= 6.1.4)
      activejob (= 6.1.4)
      activerecord (= 6.1.4)
      activestorage (= 6.1.4)
      activesupport (= 6.1.4)
      mail (>= 2.7.1)
    actionmailer (6.1.4)
      actionpack (= 6.1.4)
      actionview (= 6.1.4)
      activejob (= 6.1.4)
      activesupport (= 6.1.4)
      mail (~> 2.5, >= 2.5.4)
      rails-dom-testing (~> 2.0)
    actionpack (6.1.4)
      actionview (= 6.1.4)
      activesupport (= 6.1.4)
      rack (~> 2.0, >= 2.0.9)
      rack-test (>= 0.6.3)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.0, >= 1.2.0)
    actiontext (6.1.4)
      actionpack (= 6.1.4)
      activerecord (= 6.1.4)
      activestorage (= 6.1.4)
      activesupport (= 6.1.4)
      nokogiri (>= 1.8.5)
    actionview (6.1.4)
      activesupport (= 6.1.4)
      builder (~> 3.1)
      erubi (~> 1.4)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.1, >= 1.2.0)
    activejob (6.1.4)
      activesupport (= 6.1.4)
      globalid (>= 0.3.6)
    activemodel (6.1.4)
      activesupport (= 6.1.4)
    activerecord (6.1.4)
      activemodel (= 6.1.4)
      activesupport (= 6.1.4)
    activestorage (6.1.4)
      actionpack (= 6.1.4)
      activejob (= 6.1.4)
      activerecord (= 6.1.4)
      activesupport (= 6.1.4)
      marcel (~> 1.0.0)
      mini_mime (>= 1.1.0)
    activesupport (6.1.4)
      concurrent-ruby (~> 1.0, >= 1.0.2)
      i18n (>= 1.6, < 2)
      minitest (>= 5.1)
      tzinfo (~> 2.0)
      zeitwerk (~> 2.3)
    bcrypt (3.1.16)
    bootsnap (1.7.5)
      msgpack (~> 1.0)
    builder (3.2.4)
    byebug (11.1.3)
    concurrent-ruby (1.1.9)
    crass (1.0.6)
    devise (4.8.0)
      bcrypt (~> 3.0)
      orm_adapter (~> 0.1)
      railties (>= 4.1.0)
      responders
      warden (~> 1.2.3)
    devise-jwt (0.8.1)
      devise (~> 4.0)
      warden-jwt_auth (~> 0.5)
    dotenv (2.7.6)
    dotenv-rails (2.7.6)
      dotenv (= 2.7.6)
      railties (>= 3.2)
    dry-auto_inject (0.8.0)
      dry-container (>= 0.3.4)
    dry-configurable (0.12.1)
      concurrent-ruby (~> 1.0)
      dry-core (~> 0.5, >= 0.5.0)
    dry-container (0.8.0)
      concurrent-ruby (~> 1.0)
      dry-configurable (~> 0.1, >= 0.1.3)
    dry-core (0.7.1)
      concurrent-ruby (~> 1.0)
    erubi (1.10.0)
    ffi (1.15.3)
    globalid (0.4.2)
      activesupport (>= 4.2.0)
    i18n (1.8.10)
      concurrent-ruby (~> 1.0)
    jbuilder (2.11.2)
      activesupport (>= 5.0.0)
    jwt (2.2.3)
    listen (3.5.1)
      rb-fsevent (~> 0.10, >= 0.10.3)
      rb-inotify (~> 0.9, >= 0.9.10)
    loofah (2.10.0)
      crass (~> 1.0.2)
      nokogiri (>= 1.5.9)
    mail (2.7.1)
      mini_mime (>= 0.1.1)
    marcel (1.0.1)
    method_source (1.0.0)
    mini_mime (1.1.0)
    minitest (5.14.4)
    msgpack (1.4.2)
    nio4r (2.5.7)
    nokogiri (1.11.7-x86_64-darwin)
      racc (~> 1.4)
    orm_adapter (0.5.0)
    pg (1.2.3)
    puma (5.3.2)
      nio4r (~> 2.0)
    pundit (2.1.0)
      activesupport (>= 3.0.0)
    racc (1.5.2)
    rack (2.2.3)
    rack-cors (1.1.1)
      rack (>= 2.0.0)
    rack-test (1.1.0)
      rack (>= 1.0, < 3)
    rails (6.1.4)
      actioncable (= 6.1.4)
      actionmailbox (= 6.1.4)
      actionmailer (= 6.1.4)
      actionpack (= 6.1.4)
      actiontext (= 6.1.4)
      actionview (= 6.1.4)
      activejob (= 6.1.4)
      activemodel (= 6.1.4)
      activerecord (= 6.1.4)
      activestorage (= 6.1.4)
      activesupport (= 6.1.4)
      bundler (>= 1.15.0)
      railties (= 6.1.4)
      sprockets-rails (>= 2.0.0)
    rails-dom-testing (2.0.3)
      activesupport (>= 4.2.0)
      nokogiri (>= 1.6)
    rails-html-sanitizer (1.3.0)
      loofah (~> 2.3)
    railties (6.1.4)
      actionpack (= 6.1.4)
      activesupport (= 6.1.4)
      method_source
      rake (>= 0.13)
      thor (~> 1.0)
    rake (13.0.6)
    rb-fsevent (0.11.0)
    rb-inotify (0.10.1)
      ffi (~> 1.0)
    responders (3.0.1)
      actionpack (>= 5.0)
      railties (>= 5.0)
    spring (2.1.1)
    sprockets (4.0.2)
      concurrent-ruby (~> 1.0)
      rack (> 1, < 3)
    sprockets-rails (3.2.2)
      actionpack (>= 4.0)
      activesupport (>= 4.0)
      sprockets (>= 3.0.0)
    thor (1.1.0)
    tzinfo (2.0.4)
      concurrent-ruby (~> 1.0)
    warden (1.2.9)
      rack (>= 2.0.9)
    warden-jwt_auth (0.5.0)
      dry-auto_inject (~> 0.6)
      dry-configurable (~> 0.9)
      jwt (~> 2.1)
      warden (~> 1.2)
    websocket-driver (0.7.5)
      websocket-extensions (>= 0.1.0)
    websocket-extensions (0.1.5)
    zeitwerk (2.4.2)

PLATFORMS
  x86_64-darwin-20

DEPENDENCIES
  bootsnap (>= 1.4.4)
  byebug
  devise
  devise-jwt
  dotenv-rails
  jbuilder (~> 2.7)
  listen (~> 3.3)
  pg (~> 1.1)
  puma (~> 5.0)
  pundit
  rack-cors
  rails (~> 6.1.3, >= 6.1.3.2)
  spring
  tzinfo-data

RUBY VERSION
   ruby 2.6.6p146

BUNDLED WITH
   2.2.22
```

--- TEMPLATE END ----------------------------------------------------------------

### Did you try upgrading rubygems & bundler?

I did, I got a similar permission error when I ran `bundle update --bundler`. The error:
--- ERROR REPORT TEMPLATE -------------------------------------------------------

```
Errno::EACCES: Permission denied @ dir_s_mkdir - /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/extensions/x86_64-darwin-19/2.6.0/websocket-driver-0.7.5
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/2.6.0/fileutils.rb:239:in `mkdir'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/2.6.0/fileutils.rb:239:in `fu_mkdir'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/2.6.0/fileutils.rb:217:in `block (2 levels) in mkdir_p'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/2.6.0/fileutils.rb:215:in `reverse_each'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/2.6.0/fileutils.rb:215:in `block in mkdir_p'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/2.6.0/fileutils.rb:200:in `each'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/2.6.0/fileutils.rb:200:in `mkdir_p'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/rubygems/ext/builder.rb:217:in `write_gem_make_out'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/rubygems/ext/builder.rb:145:in `build_error'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/rubygems/ext/builder.rb:179:in `rescue in build_extension'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/rubygems/ext/builder.rb:168:in `build_extension'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/rubygems/ext/builder.rb:205:in `block in build_extensions'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/rubygems/ext/builder.rb:202:in `each'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/rubygems/ext/builder.rb:202:in `build_extensions'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/rubygems/installer.rb:851:in `build_extensions'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/bundler/rubygems_gem_installer.rb:71:in `build_extensions'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/bundler/rubygems_gem_installer.rb:28:in `install'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/bundler/source/rubygems.rb:207:in `install'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/bundler/installer/gem_installer.rb:54:in `install'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/bundler/installer/gem_installer.rb:16:in `install_from_spec'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/bundler/installer/parallel_installer.rb:186:in `do_install'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/bundler/installer/parallel_installer.rb:177:in `block in worker_pool'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/bundler/worker.rb:62:in `apply_func'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/bundler/worker.rb:57:in `block in process_queue'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/bundler/worker.rb:54:in `loop'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/bundler/worker.rb:54:in `process_queue'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/bundler/worker.rb:91:in `block (2 levels) in create_threads'
Errno::EACCES: Permission denied @ dir_s_mkdir - /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/extensions/x86_64-darwin-19/2.6.0/websocket-driver-0.7.5
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/2.6.0/fileutils.rb:239:in `mkdir'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/2.6.0/fileutils.rb:239:in `fu_mkdir'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/2.6.0/fileutils.rb:217:in `block (2 levels) in mkdir_p'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/2.6.0/fileutils.rb:215:in `reverse_each'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/2.6.0/fileutils.rb:215:in `block in mkdir_p'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/2.6.0/fileutils.rb:200:in `each'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/2.6.0/fileutils.rb:200:in `mkdir_p'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/rubygems/ext/builder.rb:169:in `build_extension'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/rubygems/ext/builder.rb:205:in `block in build_extensions'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/rubygems/ext/builder.rb:202:in `each'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/rubygems/ext/builder.rb:202:in `build_extensions'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/rubygems/installer.rb:851:in `build_extensions'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/bundler/rubygems_gem_installer.rb:71:in `build_extensions'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/bundler/rubygems_gem_installer.rb:28:in `install'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/bundler/source/rubygems.rb:207:in `install'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/bundler/installer/gem_installer.rb:54:in `install'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/bundler/installer/gem_installer.rb:16:in `install_from_spec'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/bundler/installer/parallel_installer.rb:186:in `do_install'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/bundler/installer/parallel_installer.rb:177:in `block in worker_pool'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/bundler/worker.rb:62:in `apply_func'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/bundler/worker.rb:57:in `block in process_queue'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/bundler/worker.rb:54:in `loop'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/bundler/worker.rb:54:in `process_queue'
  /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/site_ruby/2.6.0/bundler/worker.rb:91:in `block (2 levels) in create_threads'

```

## Environment

```
Bundler       2.3.18
  Platforms   ruby, x86_64-darwin-19
Ruby          2.6.6p146 (2020-03-31 revision 67876) [x86_64-darwin-19]
  Full Path   /Users/dareosk/.rbenv/versions/2.6.6/bin/ruby
  Config Dir  /Users/dareosk/.rbenv/versions/2.6.6/etc
RubyGems      3.3.18
  Gem Home    /Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0
  Gem Path    /Users/dareosk/.gem/ruby/2.6.0:/Users/dareosk/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0
  User Home   /Users/dareosk
  User Path   /Users/dareosk/.gem/ruby/2.6.0
  Bin Dir     /Users/dareosk/.rbenv/versions/2.6.6/bin
OpenSSL       
  Compiled    OpenSSL 1.1.1n  15 Mar 2022
  Loaded      OpenSSL 1.1.1n  15 Mar 2022
  Cert File   /Users/dareosk/.rbenv/versions/2.6.6/openssl/ssl/cert.pem
  Cert Dir    /Users/dareosk/.rbenv/versions/2.6.6/openssl/ssl/certs
Tools         
  Git         2.27.0
  RVM         not installed
  rbenv       rbenv 1.2.0
  chruby      not installed
```

## Bundler Build Metadata

```
Built At          2022-07-14
Git SHA           fc31b7f7d9
Released Version  true
```

### Gemfile

```ruby
source 'https://rubygems.org'
git_source(:github) { |repo| "https://github.com/#{repo}.git" }

ruby '2.6.6'

# Bundle edge Rails instead: gem 'rails', github: 'rails/rails', branch: 'main'
gem 'rails', '~> 6.1.3', '>= 6.1.3.2'
# Use postgresql as the database for Active Record
gem 'pg', '~> 1.1'
# Use Puma as the app server
gem 'puma', '~> 5.0'
# Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
gem 'jbuilder', '~> 2.7'
# Use Redis adapter to run Action Cable in production
# gem 'redis', '~> 4.0'
# Use Active Model has_secure_password
# gem 'bcrypt', '~> 3.1.7'

# Use Active Storage variant
# gem 'image_processing', '~> 1.2'

# Reduces boot times through caching; required in config/boot.rb
gem 'bootsnap', '>= 1.4.4', require: false

# Use Rack CORS for handling Cross-Origin Resource Sharing (CORS), making cross-origin AJAX possible
gem 'rack-cors'
gem 'devise'
gem 'devise-jwt'
gem 'pundit'

group :development, :test do
  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
  gem 'dotenv-rails'
  gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
end

group :development do
  gem 'listen', '~> 3.3'
  # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
  gem 'spring'
end

# Windows does not include zoneinfo files, so bundle the tzinfo-data gem
gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
```

### Gemfile.lock

```
GEM
  remote: https://rubygems.org/
  specs:
    actioncable (6.1.4)
      actionpack (= 6.1.4)
      activesupport (= 6.1.4)
      nio4r (~> 2.0)
      websocket-driver (>= 0.6.1)
    actionmailbox (6.1.4)
      actionpack (= 6.1.4)
      activejob (= 6.1.4)
      activerecord (= 6.1.4)
      activestorage (= 6.1.4)
      activesupport (= 6.1.4)
      mail (>= 2.7.1)
    actionmailer (6.1.4)
      actionpack (= 6.1.4)
      actionview (= 6.1.4)
      activejob (= 6.1.4)
      activesupport (= 6.1.4)
      mail (~> 2.5, >= 2.5.4)
      rails-dom-testing (~> 2.0)
    actionpack (6.1.4)
      actionview (= 6.1.4)
      activesupport (= 6.1.4)
      rack (~> 2.0, >= 2.0.9)
      rack-test (>= 0.6.3)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.0, >= 1.2.0)
    actiontext (6.1.4)
      actionpack (= 6.1.4)
      activerecord (= 6.1.4)
      activestorage (= 6.1.4)
      activesupport (= 6.1.4)
      nokogiri (>= 1.8.5)
    actionview (6.1.4)
      activesupport (= 6.1.4)
      builder (~> 3.1)
      erubi (~> 1.4)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.1, >= 1.2.0)
    activejob (6.1.4)
      activesupport (= 6.1.4)
      globalid (>= 0.3.6)
    activemodel (6.1.4)
      activesupport (= 6.1.4)
    activerecord (6.1.4)
      activemodel (= 6.1.4)
      activesupport (= 6.1.4)
    activestorage (6.1.4)
      actionpack (= 6.1.4)
      activejob (= 6.1.4)
      activerecord (= 6.1.4)
      activesupport (= 6.1.4)
      marcel (~> 1.0.0)
      mini_mime (>= 1.1.0)
    activesupport (6.1.4)
      concurrent-ruby (~> 1.0, >= 1.0.2)
      i18n (>= 1.6, < 2)
      minitest (>= 5.1)
      tzinfo (~> 2.0)
      zeitwerk (~> 2.3)
    bcrypt (3.1.16)
    bootsnap (1.7.5)
      msgpack (~> 1.0)
    builder (3.2.4)
    byebug (11.1.3)
    concurrent-ruby (1.1.9)
    crass (1.0.6)
    devise (4.8.0)
      bcrypt (~> 3.0)
      orm_adapter (~> 0.1)
      railties (>= 4.1.0)
      responders
      warden (~> 1.2.3)
    devise-jwt (0.8.1)
      devise (~> 4.0)
      warden-jwt_auth (~> 0.5)
    dotenv (2.7.6)
    dotenv-rails (2.7.6)
      dotenv (= 2.7.6)
      railties (>= 3.2)
    dry-auto_inject (0.8.0)
      dry-container (>= 0.3.4)
    dry-configurable (0.12.1)
      concurrent-ruby (~> 1.0)
      dry-core (~> 0.5, >= 0.5.0)
    dry-container (0.8.0)
      concurrent-ruby (~> 1.0)
      dry-configurable (~> 0.1, >= 0.1.3)
    dry-core (0.7.1)
      concurrent-ruby (~> 1.0)
    erubi (1.10.0)
    ffi (1.15.3)
    globalid (0.4.2)
      activesupport (>= 4.2.0)
    i18n (1.8.10)
      concurrent-ruby (~> 1.0)
    jbuilder (2.11.2)
      activesupport (>= 5.0.0)
    jwt (2.2.3)
    listen (3.5.1)
      rb-fsevent (~> 0.10, >= 0.10.3)
      rb-inotify (~> 0.9, >= 0.9.10)
    loofah (2.10.0)
      crass (~> 1.0.2)
      nokogiri (>= 1.5.9)
    mail (2.7.1)
      mini_mime (>= 0.1.1)
    marcel (1.0.1)
    method_source (1.0.0)
    mini_mime (1.1.0)
    minitest (5.14.4)
    msgpack (1.4.2)
    nio4r (2.5.7)
    nokogiri (1.11.7-x86_64-darwin)
      racc (~> 1.4)
    orm_adapter (0.5.0)
    pg (1.2.3)
    puma (5.3.2)
      nio4r (~> 2.0)
    pundit (2.1.0)
      activesupport (>= 3.0.0)
    racc (1.5.2)
    rack (2.2.3)
    rack-cors (1.1.1)
      rack (>= 2.0.0)
    rack-test (1.1.0)
      rack (>= 1.0, < 3)
    rails (6.1.4)
      actioncable (= 6.1.4)
      actionmailbox (= 6.1.4)
      actionmailer (= 6.1.4)
      actionpack (= 6.1.4)
      actiontext (= 6.1.4)
      actionview (= 6.1.4)
      activejob (= 6.1.4)
      activemodel (= 6.1.4)
      activerecord (= 6.1.4)
      activestorage (= 6.1.4)
      activesupport (= 6.1.4)
      bundler (>= 1.15.0)
      railties (= 6.1.4)
      sprockets-rails (>= 2.0.0)
    rails-dom-testing (2.0.3)
      activesupport (>= 4.2.0)
      nokogiri (>= 1.6)
    rails-html-sanitizer (1.3.0)
      loofah (~> 2.3)
    railties (6.1.4)
      actionpack (= 6.1.4)
      activesupport (= 6.1.4)
      method_source
      rake (>= 0.13)
      thor (~> 1.0)
    rake (13.0.6)
    rb-fsevent (0.11.0)
    rb-inotify (0.10.1)
      ffi (~> 1.0)
    responders (3.0.1)
      actionpack (>= 5.0)
      railties (>= 5.0)
    spring (2.1.1)
    sprockets (4.0.2)
      concurrent-ruby (~> 1.0)
      rack (> 1, < 3)
    sprockets-rails (3.2.2)
      actionpack (>= 4.0)
      activesupport (>= 4.0)
      sprockets (>= 3.0.0)
    thor (1.1.0)
    tzinfo (2.0.4)
      concurrent-ruby (~> 1.0)
    warden (1.2.9)
      rack (>= 2.0.9)
    warden-jwt_auth (0.5.0)
      dry-auto_inject (~> 0.6)
      dry-configurable (~> 0.9)
      jwt (~> 2.1)
      warden (~> 1.2)
    websocket-driver (0.7.5)
      websocket-extensions (>= 0.1.0)
    websocket-extensions (0.1.5)
    zeitwerk (2.4.2)

PLATFORMS
  x86_64-darwin-20

DEPENDENCIES
  bootsnap (>= 1.4.4)
  byebug
  devise
  devise-jwt
  dotenv-rails
  jbuilder (~> 2.7)
  listen (~> 3.3)
  pg (~> 1.1)
  puma (~> 5.0)
  pundit
  rack-cors
  rails (~> 6.1.3, >= 6.1.3.2)
  spring
  tzinfo-data

RUBY VERSION
   ruby 2.6.6p146

BUNDLED WITH
   2.2.22
```

--- TEMPLATE END ----------------------------------------------------------------

**Comment by dareosK** [comment]
Reinstalled everything, rbenv, ruby, rails and it worked

**Comment by deivid-rodriguez** [comment]
Ok, let me leave this opened though, because I have an idea of what the issue is and I want to verify it.

**Comment by dareosK** [comment]
Of course :) looking forward to your reply

**Comment by deivid-rodriguez** [comment]
Yay, confirmed, it's indeed a permissions issue. Let me open a PR so that we report a proper error instead of the bug report template.

**Comment by deivid-rodriguez** [comment]
It will be fixed by #5726. From now on you will get a more helpful error that reads like this

<img width="819" alt="Captura de Pantalla 2022-07-16 a las 10 52 06" src="https://user-images.githubusercontent.com/2887858/179347943-65dcdb45-0ac0-48cc-9e14-9bde7f8f46e7.png">

**Comment by dareosK** [comment]
Thank you very much 🌸 👋
